crates/oxc_linter/src/snapshots/schema_json.snap

```diff
@@ -60,15 +60,13 @@ expression: json
       ]
     },
     "plugins": {
-      "default": [
```

Comment:
I liked that it can not be null, and the default is an prefilled array:
![oxlint-plugin-default](https://github.com/user-attachments/assets/d845e403-5e47-4d91-8185-e68fa3bbe3ca)

Reply:
i wonder whether i can represent it in json schema as non-nullable.

the problem is, if the root oxlint config is has no "plugins", we want to use the plugins from the extends (no defaults), and if the root oxlint config does have plugins we want to merge. although maybe this is not correct, and even if extends exists, we should apply defaults.

thoughts?

Reply:
I think we can solve the JSON schema problem on its own. I found some code in this project where some custom logic is implemented.
I do not see a problem for `oxlint`, but with `oxc_language_server`. (including the other comment)
Every `.oxlintrc.json` config should be presented as an own `oxlint` project.
When we found the first `.oxlintrc.json` we use this as a baseline and do not search the parents directories.

I did not make many thought, but I think we should define an own `RootOxlintrc` structure where all configuration starts to be resolved.
Maybe we do want to support a specific oxlint config key, I think the rust code should reflect that.

Reply:
@camchenry you implemented the nested configs. Do you have any thoughts how the language server should handle this at the end?

Reply:
so if we, when merging, make all the paths absolute, and combind them, it should be ok?

Reply:
> You can see this when looking into the editors/vscode/fixtures code. They should be ignored by editor/vscode/.oxlintrc.json. But they still lint in the editor. Copying the file into the root of the project will not resolve this problem.

If i do:
```
$ cd ediors/vscode
$ npx oxlint@latest .
Found 0 warnings and 0 errors.
Finished in 4ms on 12 files using 12 threads.
```

which is expected right? so that feels like a bug specific to language server?

Reply:
Running `npx` oxlint` on the root will show the errors in CLI too.
Will create a new issue because this is an offtopic 👍